File: screenshot-toggler-plugin/ios/ScreenshotToggler.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(ScreenshotToggler, NSObject)
RCT_EXTERN_METHOD(toggleScreenshot:(BOOL)enable
                  resolver:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)
@end

File: screenshot-toggler-plugin/ios/ScreenshotToggler.swift
import Foundation
import UIKit
import React

@objc(ScreenshotToggler)
class ScreenshotToggler: NSObject {
  @objc
  func toggleScreenshot(_ enable: Bool,
                        resolver resolve: @escaping RCTPromiseResolveBlock,
                        rejecter reject: @escaping RCTPromiseRejectBlock) {
    DispatchQueue.main.async {
      resolve("iOS screenshot control is simulated (no native FLAG_SECURE)")
    }
  }

  @objc
  static func requiresMainQueueSetup() -> Bool {
    return true
  }
}
